### ISSUE TITLE
MATLAB 2018a and Undefined function FeatureDetector_ in cv.FeatureDetector  (mexopencv-3.3.0)

### ISSUE THREAD

**Issue by BhardwajShubham** [kyamagu/mexopencv#411]
Hi,
I am trying to run a code which makes use of `cv.FeatureDetector`. 

The following error is thrown
```
Undefined function or variable 'FeatureDetector_'.

Error in cv.FeatureDetector (line 61)
            this.id = FeatureDetector_(0, 'new', detectorType, varargin{:})
```
The function is indeed not defined under `cv.FeatureDetector.m` available at 
`\mexopencv-3.3.0\mexopencv-3.3.0\+cv\`
but can be located as inside` \mexopencv-3.3.0\mexopencv-3.3.0\src\+cv\private` as
`FeatureDetector_.cpp.`

I have tried including this file at 
`\mexopencv-3.3.0\mexopencv-3.3.0\src\+cv` to solve the problem but it persists.

Have tried adding the file at other possible locations but the error persists.

Any clues on how to resolve this ?

If more info required, please let me know.

**Comment by BhardwajShubham** [comment]
Also, I am using opencv-3.3.0-vc14 . So don't think version conflict may be an issue.

**Comment by amroamroamro** [comment]
[MEX-files](https://www.mathworks.com/help/matlab/matlab_external/introducing-mex-files.html) must be compiled first (the .cpp files are implemented in C++). First make sure you have a [supported compiler](https://www.mathworks.com/support/compilers.html) (Visual Studio).

As explained in the [readme file](https://github.com/kyamagu/mexopencv/blob/master/README.markdown#windows), you would run the following in MATLAB:
```matlab
>> mexopencv.make('opencv_path','C:\OpenCV\build')
```
replacing the path above with the correct location where opencv is installed.

There's additional info in the [wiki](https://github.com/kyamagu/mexopencv/wiki/Installation-%28Windows%2C-MATLAB%2C-OpenCV-3%29), especially if you want to enable *opencv_contrib* modules support (these are not included in the official `opencv-3.3.0-vc14.exe` binaries, but must be compiled from source).

**Comment by BhardwajShubham** [comment]
Thanks for this. I realized that late and have already steps taken in this direction. Will get back if there are further issues.
Thanks again.
